### PR TITLE
Add promise support to graphql-anywhere

### PR DESCRIPTION
**Pull request by JakeDawkins**
- This PR was opened on the old `graphql-anywhere` repo [here](https://github.com/apollographql/graphql-anywhere/pull/51)

* All tests here for me look broken with the error `Cannot find module 'apollo-utilities' from 'graphql.ts'`.

### Checklist:
- [x] Make sure all of the significant new logic is covered by tests

### CODE REVIEW

**Comment by carinakoo** [comment]
@JakeDawkins I dropped this into my project which consumes `graphql-anywhere` via `apollo-link-state`. Got the following error:
```
Unhandled (in react-apollo) Error: Network error: Error writing result to store for query 
  query ShowFeedbackModalQuery {
    showFeedbackModal @client
  }
/nCannot read property 'showFeedbackModal' of undefined
```
The error can be fixed by adding a `return` to the beginning of line 134 in `graphql.ts`

Thanks for working on this! Let me know if I can do anything else!

File: packages/graphql-anywhere/src/graphql.ts [hunk 1]
```diff
@@ -240,3 +295,23 @@ function merge(dest, src) {
...
+function promiseOrImmediate(obj, fn) {
+  if (isPromise(obj)) {
+    return obj.then(fn);
+  } else {
+    return fn(obj);
+  }
+}
+
+function arrayOrPromise(arr) {
+  if (arr.some(isPromise)) {
```
Comment (stubailo) [inline packages/graphql-anywhere/src/graphql.ts]: This doesn't seem to correctly handle multidimensional/nested arrays

File: packages/graphql-anywhere/src/graphql.ts [hunk 2]
```diff
@@ -87,13 +93,20 @@ export function graphql(
   // Default matcher always matches all fragments
   const fragmentMatcher = execOptions.fragmentMatcher || (() => true);
 
+  // Default deferrable is a promise
```
Comment (stubailo) [inline packages/graphql-anywhere/src/graphql.ts]: Do we actually need the ability to pass an alternate deferrable? Are we expecting people to be using like a jQuery deferrable or something?

**Comment by jbaxleyiii** [comment]
@JakeDawkins I ended up doing this another way with a duplicate but async only version. Thanks for trying to revive this!

**Pull request closed without merging**